.changeset/six-crabs-give.md

```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/shopify-api': major
+---
+
+The logger is now synchronous. This removes unnecessary `await`'s from functions that use the logger that functionally don't need to `await` anything else.
```

Comment:
I think we should provide a little more context here, mention that:
- only apps that implement their own loggers are affected
- you can still use async logger functions, you'll just need to handle them as promises
- `addHandlers` also becomes synchronous